### ISSUE TITLE
Reject pushes if subdataset is not already present on OpenNeuro

### ISSUE THREAD

**Issue by nellh** [OpenNeuroOrg/openneuro#1954]
**Is your feature request related to a problem? Please describe.**
A push containing a subdataset should verify the subdataset is present on OpenNeuro. This avoids incomplete datasets from being available.

**Describe the solution you'd like**
Any push which would create a subdataset in the draft head or a snapshot should be rejected. Maybe scan full history if this check is efficient enough.

**Describe alternatives you've considered**
Allowing unreachable subdatasets might be useful?

**Additional context**
From the testing plan for DataLad push support.

**Comment by effigies** [comment]
Here's a case for an unreachable subdataset:

```
ds000XYZ-fmriprep/
    code/
        containers// -> https://datasets.datalad.org/?dir=/repronim/containers
    sourcedata/
        ds000XYZ// -> ///openneuro/ds000XYZ
        ds000XYZ-freesurfer// -> ///openneuro/dsABCDEF
    ...
```

This would allow the Singularity image that created the derivative to be referenced by the derivative, as well as the source dataset, which is good reproducible practice and means ON doesn't have to concern itself with hosting containers to allow users to do this.

**Comment by nellh** [comment]
Yeah, I wonder if we should try to warn on submodules that OpenNeuro can't resolve but not actually reject them. We can examine .gitmodules during the pre-receive hook but in many cases we can't look at the tree even if it is present on OpenNeuro without crossing the network due to how we shard workers.

Maybe the better task is just showing the submodule information and linking to it on the dataset file tree so it's not an opaque directory entry when it can't be directly accessed by OpenNeuro.

**Comment by effigies** [comment]
> Maybe the better task is just showing the submodule information and linking to it on the dataset file tree so it's not an opaque directory entry when it can't be directly accessed by OpenNeuro.

That makes sense to me.

A related question is whether we want to reject local submodules (`url` beginning with `file:///` or `/` or `.`). Might be a bit extreme only to permit `https://`/`http://` `url`s?

**Comment by effigies** [comment]
Private data is another use-case for local and/or inaccessible submodules. This could include raw, identifiable subject data stored in `sourcedata/` of a standard BIDS dataset, or licenses used to run software that will not run without them.

We don't want to reject these for unreachable or local URLs, although we can insist that they be in BIDS-appropriate locations like `sourcedata/` or `code/`.
